Add database

Evg03 opened the pull request.

Evg03 committed:
feat: реализовать хранение объектов в БД

Evg03 committed:
test: добавить интеграционные тесты для DAO объектов

kvach2avl commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmServiceImpl.java:
После добавления БД в сервисе и/или соответствующем хранилище должна измениться реализация метода поиска популярных фильмов (getMostPopularFilms).
Сейчас используется реализация предыдущего спринта, в которой всё содержимое БД будет выгружаться в приложение и только потом будут отбрасываться лишние. Это годится, если все фильмы хранятся в памяти, но совершенно неприемлемо для БД

kvach2avl commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,46 @@
+DROP TABLE IF EXISTS mpa, films, users, friends, likes, genres, film_genres CASCADE;
+
+CREATE table IF NOT EXISTS mpa (
```
Если только что сделали drop table, то делать "IF NOT EXISTS" при создании в значительной степени бесполезно.
Такая проверка имеет смысл в том случае, если скрипт для удаления вынесен отдельно и факт его исполнения неизвестен

kvach2avl reviewed (commented):
Отличная работа, однако идеологию поиска фильмов нужно скорректировать с учетом работы с БД

Evg03 committed:
fix: исправить реализацию метода getMostPopularFilms

исправить ошибку при которой в методе getMostPopularFilms сперва выгружались из базы все фильмы, а затем происходила сортировка

Evg03 committed:
test: добавить интеграционные тесты для getMostPopularFilms

kvach2avl reviewed (commented):
ок
